CONTRIBUTING.md

```diff
@@ -470,6 +498,33 @@ Described below is an alternative way to pull changes from the official reposito
    sync with the upstream (i.e. official) repository. Hurray!
 
 
+## Advanced: pulling in changes from another branch without going through the central repository
+
+  Imagine that somebody has written a piece of code that you want to use, but they are not yet ready to create a pull request to the main oomph-lib development branch. Obviously, they could just send you the code, but the better way is to ask them to create a pull request to a branch on your forked copy of the repository. Ideally, you should create a new branch specifically to pull in the changes, but you could also pull into the development branch. You should follow the steps described in [Code review](#code-review) to pull those changes into your branch. You now have the problem that your upstream repository is different from the local copy on your machine, but you can solve this problem with the git merge command.
```

Comment:
There's an even better way of doing this that avoids the other person having to create pull requests for you -- you just grab the code yourself. The process consists of just a few steps:

* Add your friend's repository as a new "remote"
   ```bash
   git remote add annoying_coworker https://github.com/MatthiasHeilManchester/oomph-lib.git
   ```
* Fetch the upstream content to see what they've been working on
   ```bash
   git fetch annoying_coworker
   ```
* If they have developed the new code on a new branch (e.g. `feature-magical-fluid-stuff`), you can just pull that branch in and it will create a branch in your local repository of the same name:
   ```bash
   git checkout --track annoying_coworker/feature-magical-fluid-stuff
   ```
   Alternatively, you can pull it into your local repository under your own name:
   ```bash
   git checkout -b feature-maths-codswallop annoying_coworker/feature-magical-fluid-stuff
   ```
   which will create the branch `feature-maths-codswallop` locally and track `annoying_coworker/feature-magical-fluid-stuff`.
* If they haven't developed the code on a new branch (which they should have done!), you will want to use the latter command, so that you don't pollute your `development` branch.
* You can then push this new branch to your GitHub repository as normal:
   ```bash
   git push -u origin feature-maths-codswallop
   ```

**Note:** The newly-created local branch will track the changes from `annoying_coworker`, so when they make any changes, you can pull them in using:
```bash
git checkout feature-maths-codswallop
git pull
```

Reply:
I like Puneet's version (with detailed commands; always good to be able to just copy and paste.

Reply:
Not convinced that this is actually better. You would have to talk to the co-worker to find out where the code it located and if you want a specific feature then they would have to create a branch from a common base (i.e. your version of the code). Doesn't it make more sense for them to add the feature to your code (pull request) rather than grabbing the code from them directly given that there might have been many other changes in their repository meaning that you might have to include various other non-essential bits. Happy to add specific commands to this bit. Can also add Puneet's suggestions, but I'm not sure that this is a workflow that we want to encourage.